lib/fs.js

```diff
@@ -395,12 +395,32 @@ function readFileAfterClose(err) {
   else
     buffer = context.buffer;
 
-  if (context.encoding)
-    buffer = buffer.toString(context.encoding);
+  if (err) return callback(err, buffer);
 
-  callback(err, buffer);
+  if (context.encoding) {
+    const result = tryToString(buffer, context.encoding);
+    if (result.error)
+      return callback(result.error, buffer);
```

Comment:
In the error case, I'd much rather the second argument be undefined as opposed to it being a different type from the success case in line 405. Perhaps something like, `callback(err,undefined,buffer)`, and in the success case, `callback(null, result.buffer, buffer)`, so that the second and third arguments are always consistent?